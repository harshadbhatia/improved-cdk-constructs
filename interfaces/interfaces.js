"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW50ZXJmYWNlcy5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbImludGVyZmFjZXMudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IiIsInNvdXJjZXNDb250ZW50IjpbImV4cG9ydCBpbnRlcmZhY2UgQ0RLRW52IHtcbiAgICBhY2NvdW50OiBzdHJpbmc7XG4gICAgcmVnaW9uOiBzdHJpbmc7XG4gIH0iXX0=