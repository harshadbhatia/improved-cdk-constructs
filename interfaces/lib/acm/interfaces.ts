export interface ACMStackConfig {
  stackName: string;
  stackDescription: string;

  acms?: ACMConfig[];

  cdnAcms?: string[]
}

export interface ACMConfig {
  name: string;
  certificateName: string;
  subjectAlternativeDomainNames: string[];
  hostedZoneDomainName: string
  global: boolean
}