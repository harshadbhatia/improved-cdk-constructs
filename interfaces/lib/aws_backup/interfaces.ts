export interface BackupStackConfig {
  stackName: string;
  backups: BackupConfig[];
}

export interface BackupConfig {
  name: string;
  rule_schedule: string;
}