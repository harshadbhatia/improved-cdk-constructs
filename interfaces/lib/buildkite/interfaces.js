"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW50ZXJmYWNlcy5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbImludGVyZmFjZXMudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IiIsInNvdXJjZXNDb250ZW50IjpbImV4cG9ydCBpbnRlcmZhY2UgQnVpbGRraXRlU3RhY2tDb25maWcge1xuICAgIHN0YWNrTmFtZTogc3RyaW5nO1xuICAgIHN0YWNrRGVzY3JpcHRpb246IHN0cmluZztcbiAgICByZXBvc2l0b3J5TmFtZTogc3RyaW5nO1xufSJdfQ==