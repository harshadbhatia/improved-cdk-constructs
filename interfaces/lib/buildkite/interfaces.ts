export interface BuildkiteStackConfig {
    stackName: string;
    stackDescription: string;
    repositoryName: string;
}