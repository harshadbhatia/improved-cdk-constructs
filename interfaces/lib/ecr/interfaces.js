"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW50ZXJmYWNlcy5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbImludGVyZmFjZXMudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IiIsInNvdXJjZXNDb250ZW50IjpbImV4cG9ydCBpbnRlcmZhY2UgRUNSQ2ZnIHtcbiAgICBzdGFja05hbWU6IHN0cmluZztcbiAgICBzdGFja0Rlc2NyaXB0aW9uOiBzdHJpbmc7XG5cbiAgICByZXBvczogRUNSUmVwb1tdXG5cbiAgICBleGlzdGluZ1JlcG9zOiBFQ1JSZXBvW11cbn1cblxuZXhwb3J0IGludGVyZmFjZSBFQ1JSZXBvIHtcbiAgICByZXBvc2l0b3J5TmFtZTogc3RyaW5nO1xuICAgIHJ1bGVzPzogRUNSUnVsZVtdO1xuICAgIGxhbWJkYUNvbnRhaW5lcj86IGJvb2xlYW47XG4gICAgYWxsb3dBY2NvdW50QWNjZXNzOiBzdHJpbmdbXTtcbn1cblxuZXhwb3J0IGludGVyZmFjZSBFQ1JSdWxlIHtcbiAgICB0YWdQcmVmaXg6IHN0cmluZ1tdO1xuICAgIG1heEFnZTogbnVtYmVyO1xuICAgIG1heEltYWdlQ291bnQ6IG51bWJlcjtcbn0iXX0=