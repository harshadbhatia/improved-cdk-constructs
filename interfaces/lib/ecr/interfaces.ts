import { cdk } from "../../../deps.ts";


export interface ECRStackProps extends cdk.StackProps {
    stackName: string;
    stackDescription: string;

    repos: ECRRepo[]

    existingRepos: ECRRepo[]
}

export interface ECRRepo {
    repositoryName: string;
    rules?: ECRRule[];
    lambdaContainer?: boolean;
    allowAccountAccess: string[];
}

export interface ECRRule {
    tagPrefix: string[];
    maxAge: number;
    maxImageCount: number;
}