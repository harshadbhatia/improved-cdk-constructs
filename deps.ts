import cdk from 'https://esm.sh/aws-cdk-lib@2.49.1'

import cdk_assertions from 'https://esm.sh/aws-cdk-lib@2.49.1/assertions'
import ec2 from 'https://esm.sh/aws-cdk-lib@2.49.1/aws-ec2'
import ecr from 'https://esm.sh/aws-cdk-lib@2.49.1/aws-ecr'
import eks from 'https://esm.sh/aws-cdk-lib@2.49.1/aws-eks'
import ssm from 'https://esm.sh/aws-cdk-lib@2.49.1/aws-ssm'
import iam from 'https://esm.sh/aws-cdk-lib@2.49.1/aws-iam'
import s3 from 'https://esm.sh/aws-cdk-lib@2.49.1/aws-s3'
import cloudfront from 'https://esm.sh/aws-cdk-lib@2.49.1/aws-cloudfront'
export * as constructs from "https://esm.sh/constructs@10.1.146"

export { cdk, ec2, ssm, ecr, iam, s3, cloudfront, eks }

export { cdk_assertions }
