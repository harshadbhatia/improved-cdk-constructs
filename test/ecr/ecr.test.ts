import { assertSnapshot } from "https://deno.land/std@0.161.0/testing/snapshot.ts";
import { cdk, cdk_assertions } from "../../deps.ts";
import { VPCConfig } from "../../interfaces/lib/vpc/interfaces.ts";
import { yamlCfgFileReader } from "../../lib/utils/file-reader.ts";

// Deno.test('VPC Dummy test', () => {
//     const app = new cdk.App();
//     const cfg = yamlCfgFileReader('vpc/vpc.yaml', '../../examples/') as VPCConfig
//     const stack = new VPCStack(app, 'MyTestVPCStack', cfg)
//     const actual = JSON.stringify(app.synth().getStackArtifact(stack.artifactId).template);
//     // assertStringIncludes(actual, 'AWS::SNS::Topic')

// });

Deno.test("isECRSnapshotMatch", async function (t): Promise<void> {
    const app = new cdk.App();
    const cfg = yamlCfgFileReader('ecr/ecr.yaml', '../../examples/') as VPCConfig

    const stack = new ECRStack(app, 'MyECRStack', cfg)
    const template = cdk_assertions.Template.fromStack(stack);

    await assertSnapshot(t, template.toJSON());
});