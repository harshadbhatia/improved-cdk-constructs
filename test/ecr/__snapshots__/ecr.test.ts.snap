export const snapshot = {};

snapshot[`isECRSnapshotMatch 1`] = `
{
  Parameters: {
    BootstrapVersion: {
      Default: "/cdk-bootstrap/hnb659fds/version",
      Description: "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      Type: "AWS::SSM::Parameter::Value<String>",
    },
  },
  Resources: {
    Repoecrrepo13414714E: {
      DeletionPolicy: "Retain",
      Properties: {
        RepositoryName: "ecr-repo-1",
        RepositoryPolicyText: {
          Statement: [
            {
              Action: [
                "ecr:GetDownloadUrlForLayer",
                "ecr:BatchGetImage",
                "ecr:BatchCheckLayerAvailability",
                "ecr:PutImage",
                "ecr:InitiateLayerUpload",
                "ecr:UploadLayerPart",
                "ecr:CompleteLayerUpload",
              ],
              Effect: "Allow",
              Principal: {
                AWS: [
                  {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          Ref: "AWS::Partition",
                        },
                        ":iam::123456323232:root",
                      ],
                    ],
                  },
                  {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          Ref: "AWS::Partition",
                        },
                        ":iam::133456323232:root",
                      ],
                    ],
                  },
                  {
                    "Fn::Join": [
                      "",
                      [
                        "arn:",
                        {
                          Ref: "AWS::Partition",
                        },
                        ":iam::443456323232:root",
                      ],
                    ],
                  },
                ],
              },
              Sid: "AllowAccountAccess",
            },
          ],
          Version: "2012-10-17",
        },
      },
      Type: "AWS::ECR::Repository",
      UpdateReplacePolicy: "Retain",
    },
    Repoecrrepo2E6F9F8E7: {
      DeletionPolicy: "Retain",
      Properties: {
        RepositoryName: "ecr-repo-2",
        RepositoryPolicyText: {
          Statement: [
            {
              Action: [
                "ecr:GetDownloadUrlForLayer",
                "ecr:BatchGetImage",
                "ecr:BatchCheckLayerAvailability",
                "ecr:PutImage",
                "ecr:InitiateLayerUpload",
                "ecr:UploadLayerPart",
                "ecr:CompleteLayerUpload",
              ],
              Effect: "Allow",
              Principal: {
                AWS: {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        Ref: "AWS::Partition",
                      },
                      ":iam::",
                      {
                        Ref: "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              Sid: "AllowAccountAccess",
            },
          ],
          Version: "2012-10-17",
        },
      },
      Type: "AWS::ECR::Repository",
      UpdateReplacePolicy: "Retain",
    },
  },
  Rules: {
    CheckBootstrapVersion: {
      Assertions: [
        {
          Assert: {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    Ref: "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          AssertDescription: "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
